cc/crypto/server_encryptor.cc

```diff
@@ -16,8 +16,86 @@
 
 #include "cc/crypto/server_encryptor.h"
 
+#include <optional>
+
 #include "absl/status/statusor.h"
+#include "cc/crypto/common.h"
 #include "cc/crypto/hpke/recipient_context.h"
 #include "oak_crypto/proto/v1/crypto.pb.h"
 
-namespace oak::crypto {}  // namespace oak::crypto
+namespace oak::crypto {
+
+namespace {
+using oak::crypto::v1::EncryptedRequest;
+using oak::crypto::v1::EncryptedResponse;
+}  // namespace
+
+absl::StatusOr<DecryptionResult> ServerEncryptor::Decrypt(absl::string_view encrypted_request) {
+  // Deserialize request.
+  EncryptedRequest request;
+  if (!request.ParseFromString(encrypted_request)) {
+    return absl::InvalidArgumentError("couldn't deserialize request");
+  }
+
+  // Get recipient context.
+  if (!recipient_request_context_) {
+    // Get serialized encapsulated public key.
+    if (!request.has_serialized_encapsulated_public_key()) {
+      return absl::InvalidArgumentError(
+          "serialized encapsulated public key is not present in the initial request message");
+    }
+    std::string serialized_encapsulated_public_key = request.serialized_encapsulated_public_key();
+
+    // Create recipient context.
+    absl::StatusOr<RecipientContext> setup_base_recipient_result =
+        SetupBaseRecipient(serialized_encapsulated_public_key, server_key_pair_);
+    if (!setup_base_recipient_result.ok()) {
+      return setup_base_recipient_result.status();
+    }
+    recipient_request_context_ = std::move(setup_base_recipient_result->recipient_request_context);
+    recipient_response_context_ =
+        std::move(setup_base_recipient_result->recipient_response_context);
+  }
+
+  // Decrypt request.
+  absl::StatusOr<std::string> open_result =
+      (*recipient_request_context_)
+          ->Open(request.encrypted_message().ciphertext(),
+                 request.encrypted_message().associated_data());
+  if (!open_result.ok()) {
+    return open_result.status();
+  }
+  std::string plaintext = std::move(*open_result);
+
+  return DecryptionResult{plaintext, request.encrypted_message().associated_data()};
+}
+
+absl::StatusOr<std::string> ServerEncryptor::Encrypt(absl::string_view plaintext,
+                                                     absl::string_view associated_data) {
+  // Get recipient context.
+  if (!recipient_response_context_) {
+    return absl::InternalError("server encryptor is not initialized");
+  }
+
+  // Encrypt response.
+  absl::StatusOr<std::string> seal_result =
+      (*recipient_response_context_)->Seal(plaintext, associated_data);
+  if (!seal_result.ok()) {
+    return seal_result.status();
+  }
+  std::string ciphertext = std::move(*seal_result);
```

Comment:
Instead of introducing a new variable here, stick with the 'sealed_result' variable. 

Perhaps a good change here would be to change the variable nam 'sealed_result' to just 'ciphertext'

Reply:
Done